operator/pkg/gateway-api/httproute.go

```diff
@@ -146,16 +146,28 @@ func (r *httpRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return builder.Complete(r)
 }
 
-func (r *httpRouteReconciler) hasGatewayParent() func(object client.Object) bool {
+func (r *httpRouteReconciler) hasGatewayParent(ctx context.Context, controllerName string) func(object client.Object) bool {
 	return func(obj client.Object) bool {
+		scopedLog := r.logger.With(logfields.Controller, gateway, logfields.Resource, obj.GetName())
 		hr, ok := obj.(*gatewayv1.HTTPRoute)
 		if !ok {
 			return false
 		}
 
 		for _, parent := range hr.Spec.ParentRefs {
 			if helpers.IsGateway(parent) {
-				return true
+				// check if the Gateway has the correct controller
```

Comment:
do we need to update similar logic for grpcroute and tlsroute ?

Reply:
likely yes - this logic also isn't currently working in my local setup; will revisit later today